src/layer/Tooltip.js

```diff
@@ -385,22 +385,38 @@ Layer.include({
 		return this._tooltip;
 	},
 
-	_addFocusListeners() {
+	_addFocusListeners(remove) {
 		if (this.getElement) {
-			this._addFocusListenersOnLayer(this);
-		} else if (this.eachLayer) {
-			this.eachLayer(this._addFocusListenersOnLayer, this);
+			const el = this.getElement();
+			if (el) {
+				const onOff = remove ? 'off' : 'on';
+				DomEvent[onOff](el, 'focus', () => this._handleFocus(this), this);
+				DomEvent[onOff](el, 'blur', this._handleBlur, this);
+			}
+		} else if (this.eachLayer && this._map.getRenderer(this)) {
+			this.eachLayer(layer => this._addFocusListenersOnLayer(layer), this);
 		}
 	},
 
-	_addFocusListenersOnLayer(layer) {
-		const el = typeof layer.getElement === 'function' && layer.getElement();
-		if (el) {
-			DomEvent.on(el, 'focus', function () {
-				this._tooltip._source = layer;
-				this.openTooltip();
-			}, this);
-			DomEvent.on(el, 'blur', this.closeTooltip, this);
+	_handleFocus(source) {
+		if (this._tooltip) {
+			this._tooltip._source = source;
+			this.openTooltip();
+		}
+	},
+
+	_handleBlur() {
+		this.closeTooltip();
+	},
+
+	_addFocusListenersOnLayer(layer, remove) {
+		if (layer.getElement) {
+			const el = layer.getElement();
+			if (el) {
+				const onOff = remove ? 'off' : 'on';
+				DomEvent[onOff](el, 'focus', () => this._handleFocus(layer), this);
+				DomEvent[onOff](el, 'blur', this._handleBlur, this);
+			}
 		}
```

Comment:
I think we should do the following (or something similar if there is a better way):

1. `_addFocusListeners` does the same for a element as `_addFocusListenersOnLayer`, so we don't duplicate the code.
2. Then the `_handleBlur` don't need to be a function anymore, because we can call `closeTooltip` directly
3. We store the `named` function / listener (`el._leaflet_focus_handler`) for the focus on the element, because an anonymous function / listener can't be removed with `off`

```
	_addFocusListeners(remove) {
		if (this.getElement) {
			this._addFocusListenersOnLayer(this, remove);
		} else if (this.eachLayer && this._map.getRenderer(this)) {
			this.eachLayer(layer => this._addFocusListenersOnLayer(layer, remove), this);
		}
	},

	_addFocusListenersOnLayer(layer, remove) {
		const el = typeof layer.getElement === 'function' && layer.getElement();
		if (el) {
			const onOff = remove ? 'off' : 'on';
			if (!remove) {
				// Remove focus listener, if already existing
				el._leaflet_focus_handler && DomEvent.off(el, 'focus', el._leaflet_focus_handler, this);

				el._leaflet_focus_handler = () => {
					if (this._tooltip) {
						this._tooltip._source = layer;
						this.openTooltip();
					}
				};
			}

			el._leaflet_focus_handler && DomEvent[onOff](el, 'focus', el._leaflet_focus_handler, this);
			DomEvent[onOff](el, 'blur', this.closeTooltip, this);

			if (remove) {
				delete el._leaflet_focus_handler;
			}
		}
	},
```